reconcile/openshift_namespace_labels.py

```diff
@@ -0,0 +1,395 @@
+import json
+import logging
+import sys
+from threading import Lock
+from typing import Any, Dict, Generator, List, Optional, Tuple, Union
+
+from kubernetes.client.exceptions import ApiException
+
+import reconcile.queries as queries
+import reconcile.utils.threaded as threaded
+from reconcile.utils.defer import defer
+from reconcile.utils.oc import OC_Map, OCNative, StatusCodeError
+from reconcile.utils.oc import validate_labels
+
+from reconcile.utils.sharding import is_in_shard
+from reconcile.utils.state import State
+
+_LOG = logging.getLogger(__name__)
+
+QONTRACT_INTEGRATION = 'openshift-namespace-labels'
+
+DESIRED = 'desired'
+MANAGED = 'managed'
+CURRENT = 'current'
+CHANGED = 'changed'
+UPDATED_MANAGED = 'updated-managed'
+
+Labels = Dict[str, Optional[str]]
+LabelKeys = List[str]
+LabelsOrKeys = Union[Labels, LabelKeys, None]
```

Comment:
One other minor thing to consider here is that unless `None` is really useful in this case for a non-obvious reason, it could make more sense to just use `Optional[LabelsOrKeys]` where you need to while dropping None from the union.  I'll leave that up to you, but something like:

```
def some_func() -> Optional[LabelsOrKeys]
```

is much more obvious to me that the function can return `None`.  Mypy doesn't care, just more of a consideration for people reading through the code.